src/platform/tls_openssl.c

```diff
@@ -1462,6 +1462,30 @@ CxPlatTlsSecConfigCreate(
         }
     }
 
+    if (CredConfigFlags & QUIC_CREDENTIAL_FLAG_USE_CACERTFILE) {
+        if (CredConfig->CaCertificateFile) {
+            Ret = SSL_CTX_load_verify_locations(SecurityConfig->SSLCtx,
+                                                CredConfig->CaCertificateFile,
+                                                NULL);
+            if (Ret != 1) {
+                QuicTraceEvent(
+                    LibraryErrorStatus,
+                    "[ lib] ERROR, %u, %s.",
+                    ERR_get_error(),
+                    "SSL_CTX_load_verify_locations failed");
+                Status = QUIC_STATUS_TLS_ERROR;
+                goto Exit;
+            }
+        } else {
+            QuicTraceEvent(
+                LibraryErrorStatus,
+                "QUIC_CREDENTIAL_FLAG_USE_CACERTFILE is set but "
+                "CaCertificateFile is not set.");
```

Comment:
I don't think we need to error out if the flag is set but the pointer is NULL. Let's just treat the flag as "please check the pointer". So please combine the two top-level `if` checks into one, and only do anything if both are true.